packages/superset-ui-dimension/test/getTextDimension.test.ts

```diff
@@ -0,0 +1,147 @@
+import { getTextDimension } from '../src/index';
+import addDummyFill from './addDummyFill';
+
+describe('getTextDimension(input)', () => {
+  describe('returns default dimension if getBBox() is not available', () => {
+    it('returns default value for default dimension', () => {
+      expect(
+        getTextDimension({
+          text: 'sample text',
+        }),
+      ).toEqual({
+        height: 20,
+        width: 100,
+      });
+    });
+    it('return specified value if specified', () => {
+      expect(
+        getTextDimension(
+          {
+            text: 'sample text',
+          },
+          {
+            height: 30,
+            width: 400,
+          },
+        ),
+      ).toEqual({
+        height: 30,
+        width: 400,
+      });
+    });
+  });
+  describe('returns dimension of the given text', () => {
+    let originalFn: () => DOMRect;
+
+    beforeEach(() => {
+      // @ts-ignore - fix jsdom
+      originalFn = SVGElement.prototype.getBBox;
+      addDummyFill();
+    });
+
+    afterEach(() => {
+      // @ts-ignore - fix jsdom
+      SVGElement.prototype.getBBox = originalFn;
+    });
+
+    it('takes text as argument', () => {
+      expect(
+        getTextDimension({
+          text: 'sample text',
+        }),
+      ).toEqual({
+        height: 20,
+        width: 200,
+      });
+    });
+    it('accepts provided class via className', () => {
+      expect(
+        getTextDimension({
+          text: 'sample text',
+          className: 'test-class',
+        }),
+      ).toEqual({
+        height: 20,
+        width: 100,
+      });
+    });
+    it('accepts provided style.font', () => {
+      expect(
+        getTextDimension({
+          text: 'sample text',
+          style: {
+            font: 'italic 700 30px Lobster',
+          },
+        }),
+      ).toEqual({
+        height: 30,
+        width: 1125,
```

Comment:
Would also be nice to have a comment here for how we arrived at `1125` (`250 * (30/20) * 1.5 * 2`).

Reply:
done